homeassistant/components/harmony/remote.py

```diff
@@ -325,17 +288,18 @@ async def got_connected(self, _=None):
             # We were disconnected before.
             await self.new_config()
 
+            if self._unsub_mark_disconnected:
+                self._unsub_mark_disconnected()
+
     async def got_disconnected(self, _=None):
         """Notification that we're disconnected from the HUB."""
         _LOGGER.debug("%s: disconnected from the HUB", self._name)
         self._available = False
         # We're going to wait for 10 seconds before announcing we're
         # unavailable, this to allow a reconnection to happen.
-        await asyncio.sleep(10)
-
-        if not self._available:
-            # Still disconnected. Let the state engine know.
-            self.async_write_ha_state()
+        self._unsub_mark_disconnected = async_call_later(
```

Comment:
Should switches have the same behavior on disconnection?  Maybe make this functionality a base/mix-in class for both switches and remotes?

Reply:
Originally I thought it made more sense for the switches to respond immediately (while keeping the remote behavior the same). Thinking about it more, that's pretty silly. Anyhow, I've split out a mixin to handle this and updated the remote slightly to get the availability state consistently with how the switches do it.

It turns out that the remote had to set it's availability state in a bunch of places because (even in the old code) it would subscribe to callbacks _after_ connection callbacks would be called the first time around. The data object we're using configures these callbacks prior to making the first connection so this is no longer an issue.